extensions/GitExtension/FileExplorerGitIntegration/Models/RepositoryWrapper.cs

```diff
@@ -81,13 +117,24 @@ public RepositoryWrapper(string rootFolder)
 
     private CommitLogCache GetCommitLogCache()
     {
-        // Fast path: if we have an up-to-date commit log, return that
+        var result = GitExecute.ExecuteGitCommand(_gitDetect.GitConfiguration.ReadInstallPath(), _workingDirectory, "rev-parse HEAD");
+        if (result.Status != ProviderOperationStatus.Success)
+        {
+            throw result.Ex ?? new InvalidOperationException();
+        }
+
+        string? head = result.Output?.Trim();
+        if (string.IsNullOrEmpty(head))
+        {
+            throw new InvalidOperationException("Git command output is null and/or the repository has no commits");

```

Comment:
NIT: I don't believe the command output can be null and have no commits.  Maybe remove `and`.